### PR TITLE
LWDEV-6056 Develop generic ethereum integration

### PR DESCRIPTION
**Pull request by PaterSantyago**

### CODE REVIEW

File: src/Lykke.Service.EthereumSignApi/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+        public IServiceProvider ConfigureServices(
+            IServiceCollection services)
+        {
+            return services.BuildEthereumServiceProvider<AppSettings>
+            (
+                serviceName: "SignApi",
+                
+                #if ENABLE_SENSITIVE_LOGGING
```
Comment (some-dev) [inline src/Lykke.Service.EthereumSignApi/Startup.cs:27]: In other integration any logging is disabled. Without any option.
Do ethereum require this option?

File: src/Lykke.Service.EthereumWorker.AzureRepositories/BlockIndexationLockRepository.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            return _locks.DeleteIfExistAsync
+            (
+                partitionKey: partitionKey,
+                rowKey: rowKey
+            );
+        }
+
+        public async Task<IEnumerable<BlockIndexationLock>> GetAsync()
+        {
+            return (await _locks.GetDataAsync())
```
Comment (cassade) [inline src/Lykke.Service.EthereumWorker.AzureRepositories/BlockIndexationLockRepository.cs:63]: Full table scan. Are you sure that count of entities always will be reasonably small?

File: src/Lykke.Service.EthereumWorker.Services/BlockchainService.cs [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    public class BlockchainService : BlockhainServiceBase, IBlockchainService
+    {
+        private readonly int _confirmationLevel;
+
+        private readonly string[] _valueTransferCallCodes =
+            new[] {"CREATE", "CALL", "CALLCODE", "DELEGATECALL", "SUICIDE"};
+        
+        
+        public BlockchainService(
+            ILogFactory logFactory,
```
Comment (cassade) [inline src/Lykke.Service.EthereumWorker.Services/BlockchainService.cs]: Parameter never used.

File: src/Lykke.Service.EthereumWorker.AzureRepositories/BlockIndexationLockRepository.cs (hunk 2 above)
Comment (PaterSantyago) [reply src/Lykke.Service.EthereumWorker.AzureRepositories/BlockIndexationLockRepository.cs:63]: Yes, I suppose, it always will be lower than 100

File: src/Lykke.Service.EthereumSignApi/Startup.cs (hunk 1 above)
Comment (PaterSantyago) [reply src/Lykke.Service.EthereumSignApi/Startup.cs:27]: Just to make it easier to debug in local and dev environment, logging is disabled by default, so it should not be problem in test and prod envs

**Review by cassade — APPROVED** [review]

**Pull request merged**